jenkins-pr-collector.go

```
@@ -352,63 +415,132 @@ func fetchJenkinsPluginInfo(updateCenterURL string) (map[string]PluginInfo, erro
+	if err := json.NewDecoder(resp.Body).Decode(&graphqlResp); err != nil {
+		return fmt.Errorf("failed to decode response: %v", err)
 	}
 
 	// Check for GraphQL errors
 	if len(graphqlResp.Errors) > 0 {
-		errorMsg := fmt.Sprintf("GraphQL error: %s", graphqlResp.Errors[0].Message)
-		return fmt.Errorf(errorMsg)
-	}
+		// Check if any of the errors are rate limit related
+		for _, gqlErr := range graphqlResp.Errors {
+			if isRateLimitError(fmt.Errorf(gqlErr.Message)) {
```

golangci-lint (1.62.2)

printf: non-constant format string in call to fmt.Errorf

(govet)